contracts/proxy/diamond/writable/DiamondWritableInternal.sol

```diff
@@ -122,10 +122,9 @@ abstract contract DiamondWritableInternal is IDiamondWritableInternal {
                     selectorBitIndexInSlug
                 );
 
-                // if slug is now full, write it to storage and continue with an empty slug
                 if (selectorBitIndexInSlug == 224) {
+                    // slug is now full, so write it to storage
                     l.selectorSlugs[selectorCount >> 3] = slug;
-                    slug = bytes32(0);
```

Comment:
I feel like leaving this in, would make debugging easier if we ever have another issue?

Reply:
You are right though.

Reply:
There is definitely an argument for leaving this in.  However, I want to be consistent about it.  Either we keep the out-of-bounds selectors clean at all times, or we don't bother at all.  For now, this PR takes the latter approach.

I think I'm slightly in favor of leaving the storage dirty as long as we document the behavior.  I'll add some comments.

And to be honest, I'm not sure what the current state of gas refunds is.  If we expect that a deleted slot will be filled again at some point in the future, it's not worth deleting because the future write will cost more, right?

Reply:
AFAIK, you still get 4800 refunded instead of the original 15000 (https://eips.ethereum.org/EIPS/eip-3529).

But yeah, I would say in this case it make sense to leave it dirty if we expect at some point the slot will be reused.